### PR TITLE
Address issue #255: Take Two

### PR DESCRIPTION
**Pull request by cjerdonek**
This PR addresses issue #255 and is an improved version of PR #256.

This PR doesn't yet address all of the suggestions that were made in the comments to PR #256, but I wanted to get this PR posted as soon as possible to help meet the deadline, etc.

Note that this PR doesn't DRY up the YAML data quite as much as PR #256 since PR #256 involved some manual work mixed in with the automated script.  In this PR, however, everything is automated.

An example of something not handled by the current PR is the following:

> If the field is invalid or not present, the implementation is required to ignore the ``Office`` element containing it.

In the other PR, error-handling strings like this were generated automatically using a template (aka Python format string).  But in this PR, strings like this are "hard-coded" in the YAML.  It is slightly harder to convert these automatically since the word for the parent type varies.  This DRY-ing up can be done in a subsequent PR.

Also, this PR doesn't actually _run_ the parsing, YAML generation, and new reST table generation.  That can be done in a PR immediately after this, which was something that was requested.

### CODE REVIEW

**Comment by cjerdonek** [comment]
By the way, as a check, I ran a "diff" against the YAML files in the previous PR, and it mostly checks out except for some minor differences that result from changes since then, as well as things like error strings involving the "parent" type.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -51,23 +51,103 @@ dev branch (called 'vip5' in the vip-specification's case, but it may change to
...
+[http://127.0.0.1:8000](http://127.0.0.1:8000) to see the documentation.
+
+
+## Updating Tables in the Documentation
+
+To make changes to the tables, edit the YAML files by hand as needed.
+Do not edit the reST tables by hand since they are generated automatically
+from the YAML files.
+
+Then, normalize the YAML files and update the reST tables:
```
Comment (jdmgoogle) [inline CONTRIBUTING.md]: How does one normalize the YAML files?

File: scripts/vip.py [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+_log = logging.getLogger()
+
+FORMATTER_CLASS = argparse.RawDescriptionHelpFormatter
+
+HELP_SUMMARY = """\
+Helper script for VIP contributors.
+"""
+
+def _wrap(text):
+    """Format text for help outuput."""
```
Comment (jdmgoogle) [inline scripts/vip.py]: output

**Comment by jdmgoogle** [comment]
The disclaimer here is that this is a moderate size chunk of python, and python isn't my strongest language. Down the road I'd like to see some of this code replaced with something like off-the-shelf like the docutil RST tableparser:

http://svn.python.org/projects/external/docutils-0.6/docutils/parsers/rst/tableparser.py

That's not for today, though.

If @pstenbjorn or @jungshadow (or anyone else with a good Python background) could give this a thorough once-over, I'd really appreciate that.

This gets a tentative LGTM from me.

**Comment by cjerdonek** [comment]
> Down the road I'd like to see some of this code replaced with something like off-the-shelf like the docutil RST tableparser:

The good news is that the table-parsing code can be removed from the repo _entirely_ after initially generating the YAML files (which will hopefully happen later today or else tomorrow).  The reason is that, after the YAML files are in place, we will only be _generating_ reST tables and not parsing them. :grinning:

**Comment by cjerdonek** [comment]
PR updated with typo fix.

(Also, I forgot to mention in my previous comment that the parsing code is largely confined to `parsing.py`, which is one of the improvements I made over the previous PR.  So that file can potentially be removed in its entirety after the YAML file generation.)

**Comment by jdmgoogle** [comment]
Okay. I'm all for deleting code. :)

LGTM

Cc @jungshadow @pstenbjorn @jen-tolentino

**Comment by jungshadow** [comment]
@cjerdonek I have a single issue, which is the Markdown footnote in the credits. Can you put the footnote on the same line or use superscript tags (pretty sure they're allowed)? Other than the cosmetic issue, LGTM.

**Comment by cjerdonek** [comment]
Okay, thanks, guys.  @jungshadow I added the superscript.  If that's not what you wanted, let me know, and I can change it in a later commit.  Or you can change it.

**Comment by cjerdonek** [comment]
PS - I'm going to make a new PR right now to actually run the script, etc.

**Comment by cjerdonek** [comment]
See PR #273 for part 2.

**Comment by cjerdonek** [comment]
Actually, it's probably better to leave the reST parsing code in the repo for a _little_ bit longer.  It will be useful to address @jdmgoogle's suggestion to also include in the YAML the text preceding and following each table.  That would be easy to do, and would let us get rid of a bunch of the reST files (many of which will be small after issue #255).  However, it's probably better for that to be done after this week.

**Pull request merged**